src/compiler/crystal/compiler.cr

```diff
@@ -295,7 +297,15 @@ module Crystal
 
       Crystal.timing("Codegen (linking)", @stats) do
         Dir.cd(output_dir) do
-          system %(#{CC} -o "#{output_filename}" "${@}" #{@link_flags} #{lib_flags}), object_names
+          if thin_lto
+            lto_cache_dir = "#{output_dir}/lto.cache"
+            Dir.mkdir_p(lto_cache_dir)
+            cc = ENV["CC"]? || "clang -flto=thin -Wl,-mllvm,-threads=#{n_threads},-cache_path_lto,#{lto_cache_dir},#{@release ? "-mllvm,-O2" : "-mllvm,-O0"}"
```

Comment:
If set is set by the system or to `clang-4.0` for example, then thinlto will be silently ignored.

Is it possible to detect that CC is clang >= 4.0 ?